language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -507,6 +518,32 @@ renderDecoder = \case
     DecoderRef t -> snd (genType t) <> ".decoder"
     DecoderLazy d -> "damlTypes.lazyMemo(function () { return " <> renderDecoder d <> "; })"
 
+data Encode
+    = EncodeRef TypeRef
+    | EncodeVariant [(T.Text, TypeRef)]
+    | EncodeAsIs
+    | EncodeRecord [(T.Text, TypeRef)]
+    | EncodeThrow
+
+renderEncode :: Encode -> T.Text
+renderEncode = \case
+    EncodeRef ref -> let (_, companion) = genType ref in
+        "function (__typed__) { return " <> companion <> ".encode(__typed__); }"
+    EncodeVariant alts -> T.unlines $ concat
+        [ [ "function (__typed__) {" ]
```

Comment:
Why the weird variable name?

Reply:
We don't have any hygiene support here, so it needed to be a variable name that had no chance of ever clashing with the template arguments. This is not a problem for `decoder` objects as they are objects (and thus point-free, in a sense), but for these `encode` functions shadowing could be a real issue. Happy to take any suggestion on a better name, though. This one comes from an unverified belief that `__xx__` would not be produced as a type variable name, and the fact that the `encode` functions take typed input and turn it into untyped serialized values.